File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/infrastruktur/database/statistikk/SykefraværsstatistikkIntegrasjonUtils.java
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.infrastruktur.database.statistikk;

import no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.domenemodeller.Sykefraværsstatistikk;

import java.util.List;

public interface SykefraværsstatistikkIntegrasjonUtils {

  DeleteSykefraværsstatistikkFunction getDeleteFunction();

  BatchCreateSykefraværsstatistikkFunction getBatchCreateFunction(
      List<? extends Sykefraværsstatistikk> list);
}

File: src/main/java/no/nav/arbeidsgiver/sykefravarsstatistikk/api/infrastruktur/database/statistikk/SykefraværsstatistikkIntegrasjonUtils.kt
package no.nav.arbeidsgiver.sykefravarsstatistikk.api.infrastruktur.database.statistikk

import no.nav.arbeidsgiver.sykefravarsstatistikk.api.applikasjon.domenemodeller.Sykefraværsstatistikk

interface SykefraværsstatistikkIntegrasjonUtils {
    fun getDeleteFunction(): DeleteSykefraværsstatistikkFunction
    fun getBatchCreateFunction(
        list: List<Sykefraværsstatistikk>
    ): BatchCreateSykefraværsstatistikkFunction
}
